AppDelegate.mm

```diff
@@ -96,6 +131,14 @@ - (void)applicationDidFinishLaunching:(NSNotification*)notification {
   NSDictionary* downloadURLs = [info objectForKey:@"TargetDownloadURLs"];
   NSURL* downloadURL = [NSURL URLWithString:[downloadURLs valueForKey:ARCH_KEY_NAME]];
   NSString* targetAppName = [info valueForKey:@"TargetAppName"];
+  auto* tempDir = [NSBundle.mainBundle.bundlePath stringByAppendingPathComponent:@"download"];
```

Comment:
Unzipping here might need admin rights if the installer bundle is a read-only `/Applications` directory, that would fail with this approach. We would need to use the `STPrivilegedTask` approach as a fallback.

Or perhaps we could always just run a `/bin/sh` script that unzips and does the final moves. If it fails on the first try, we can try again using `STPrivilegedTask`.